lib/stat_tracker.rb

```diff
@@ -246,6 +249,77 @@ def lowest_scoring_home_team
     end
     match.teamname
   end
-  
 
-end
+  def game_teams_by_coach
+    @game_teams.group_by do |game|
+      game.head_coach
+    end
+  end
+
+  def coach_win_percentage
+    wins_losses = {}
+    game_teams_by_coach.max_by do |coach, games|
+      wins_losses[coach] = games.map do |game|
+        game.result
+      end
+    end
+  end
+
+  def season_game_ids
+    season_game_ids = {}
+    games_by_season.map do |season, games|
+      season_game_ids[season] = games.map do |game|
+        game.game_id
+      end
+    end
+    season_game_ids
+  end
+
+  def game_team_by_season(season_id)
+    @game_teams.find_all do |row|
+      season_game_ids[season_id].include?(row.game_id)
+    end
+  end
+
+  def games_by_team_id(season_id)
+    game_team_by_season(season_id).group_by do |game|
+      game.team_id
+    end
+  end
+
+  def team_conversion_percent(season_id)
+    team_ratio = {}
+    games_by_team_id(season_id).map do |team, games|
+      goals = 0.0
+      shots = 0.0
+      games.map do |game|
+        goals += game.goals 
+        shots += game.shots
+      end
+      team_ratio[team] = goals / shots
+    end
+    team_ratio
+  end
+
+  def most_accurate_team(season_id)
+    ratio = team_conversion_percent(season_id).max_by do |team, ratio|
+      ratio
+    end
+
+    @teams.map do |team|
+      return team.teamname if team.team_id == ratio[0]
+    end
+  end
+
+
+
+  def least_accurate_team(season_id)
```

Comment:
Seeing some similarities between how Adam and I did out best and worst offense, having similar functionality apart from min / max. We will probably have to pull that out when we refactor in the future.